woocommerce/woocommerce-ios
# Stats v4: fix incorrect stats interval boundaries without using GMT dates + site time zone

Fixes #1357 
Fixes #1376

Update release notes:


## Changes
- Added `iso8601WithoutTimeZone` to format a date based on iso 8601 format without the time zone, so that the server will use the site time zone
- In Stats v4, used `iso8601WithoutTimeZone` to format the `before` and `after` dates
- Used device time zone as the default time zone so that the dates for the stats data are rendered as the value without any time zone adjustment. Removed loading site action originally for site time zone
- Updated `OrderStatsV4Interval`'s `dateStart` and `dateEnd` from GMT dates to dates in the site time zone from the API response

## Testing

- In `wp-admin` > Settings > General, set a store's time zone to UTC with an offset that is different from the device/simulator time zone
- Launch the app with stats v4 --> the data in the tabs should be consistent with the WC Admin on web
- Repeat the previous two steps with the store's time zone in a region (not UTC based)

## Example screenshots

time range | before | after
-- | -- | --
this week | ![stats_day_before](https://user-images.githubusercontent.com/1945542/67065444-ed4fc480-f1a0-11e9-98ca-4aacf4fdd96d.png) | ![stats_day_after](https://user-images.githubusercontent.com/1945542/67065443-ed4fc480-f1a0-11e9-9898-9b80b116aeb1.png)
this year | ![stats_month_before](https://user-images.githubusercontent.com/1945542/67065413-d01af600-f1a0-11e9-9978-21307c62b07d.png) | ![stats_month_after](https://user-images.githubusercontent.com/1945542/67065412-cf825f80-f1a0-11e9-81eb-f7be164e17ac.png)
